### PR TITLE
fix planner self talking issue

### PR DESCRIPTION
**Pull request by zhangxu0307**
- forbidden planner self asking
  - revise prompt issue
  - add post validation for send_to field  
- count the self talking into the internal chat round num
- remove deprecated roll_back function
- add invalid planner response in chat histroy for reference
- merge consecutive user messages

### CODE REVIEW

File: taskweaver/planner/planner.py [hunk 1]
```diff
@@ -188,6 +190,7 @@ def reply(
 
         def check_post_validity(post: Post):
             assert post.send_to is not None, "Post send_to field is None"
+            assert post.send_to != "Planner", "Post send_to field should not be Planner"
```
Comment (liqul) [inline taskweaver/planner/planner.py:193]: Should add the error message to the self-correction message

File: taskweaver/planner/planner.py [hunk 2]
```diff
@@ -211,7 +214,7 @@ def check_post_validity(post: Post):
                 f"{self.prompt_data['planner_response_schema']}"
                 "Please try to regenerate the Planner output.",
                 send_to="Planner",
-                send_from="Planner",
+                send_from="User",
```
Comment (liqul) [inline taskweaver/planner/planner.py:217]: Need to consider the need to find all the posts where the Planner is trying to correct itself. If changed to User, then it becomes hard to track.

File: taskweaver/session/session.py [hunk 3]
```diff
@@ -156,8 +155,6 @@ def _send_message(recipient: str, post: Post):
             chat_round.change_round_state("failed")
             err_message = f"Cannot process your request due to Exception: {str(e)} \n {stack_trace_str}"
             event_handler("error", err_message)
-            self.code_interpreter.rollback(chat_round)
-            self.planner.rollback(chat_round)
             self.internal_chat_num = 0
```
Comment (liqul) [inline taskweaver/session/session.py:158]: move this to finally?

File: taskweaver/planner/planner.py (hunk 1 above)
Comment (zhangxu0307) [reply taskweaver/planner/planner.py]: added

File: taskweaver/planner/planner.py (hunk 2 above)
Comment (zhangxu0307) [reply taskweaver/planner/planner.py]: revised to Planner->Planner

File: taskweaver/session/session.py (hunk 3 above)
Comment (zhangxu0307) [reply taskweaver/session/session.py]: moved

**Review by Jack-Q — APPROVED** [review]

**Review by liqul — APPROVED** [review]

**Pull request merged**